upup/pkg/fi/cloudup/openstacktasks/servergroup.go

```diff
@@ -133,14 +133,33 @@ func (_ *ServerGroup) RenderOpenstack(t *openstack.OpenstackAPITarget, a, e, cha
 		for currentLastIndex > fi.Int32Value(changes.MaxSize) {
 			iName := strings.ToLower(fmt.Sprintf("%s-%d.%s", fi.StringValue(a.IGName), currentLastIndex, fi.StringValue(a.ClusterName)))
 			instanceName := strings.Replace(iName, ".", "-", -1)
-			opts := servers.ListOpts{
-				Name: fmt.Sprintf("^%s$", instanceName),
-			}
-			instances, err := t.Cloud.ListInstances(opts)
+			// TODO: we could maybe use some filter here
+			allInstances, err := t.Cloud.ListInstances(servers.ListOpts{})
 			if err != nil {
 				return fmt.Errorf("error fetching instance list: %v", err)
 			}
 
+			instances := []servers.Server{}
+			for _, server := range allInstances {
+				val, ok := server.Metadata["k8s"]
```

Comment:
Gophercloud also supports tags in ListOpts: https://pkg.go.dev/github.com/gophercloud/gophercloud@v0.13.0/openstack/compute/v2/servers#ListOpts Can we use that at line 137 to already limit the instances?